unlock-js/src/unlockProvider.js

```diff
@@ -33,33 +33,29 @@ export default class UnlockProvider {
     return true
   }
 
-  async eth_accounts(respond) {
+  async eth_accounts() {
     // Must always return an array of addresses
     if (this.wallet) {
-      respond([this.wallet.address])
+      return [this.wallet.address]
     } else {
-      respond([])
+      return []
     }
   }
 
-  send(args, cb) {
-    const { id, jsonrpc, method } = args
-    // Calling respond with some argument will call the callback with
-    // a "fake" JSON-RPC response constructed by the provider.
-    const respond = result => {
-      cb(null, {
-        id,
-        jsonrpc,
-        result,
-      })
-    }
+  async eth_signTypedData(params) {
+    // params is [ account, data ]
+    // we don't need account
+    const data = params[1]
+    return this.wallet.signMessage(data)
```

Comment:
what is the type of `this.wallet`?  In general this seems okay but I am not able to glean a lot of context from this delta.

If this.wallet.signMessage is the method for handling erc712 signatures than it makes sense

Reply:
`this.wallet` is an ethers wallet generated from the private key, it wraps over various account-related functionality. Just wasn't 100% sure if there's actually a big difference when it comes to signing typed data, since only 2 signature methods are provided.

Reply:
I see, i'm not sure that ethers supports creating eip712 signatures. This is the most relevant issues that I found https://github.com/ethers-io/ethers.js/issues/98.

Reply:
Not sure if this is helpful, here is a link to a somewhat generic implementation of the signature method https://github.com/MetaMask/eth-sig-util